File: topico_3/Python/1-countNeg.py
# Conte os números negativos de uma lista

def countNeg(xs):
    cont = 0
    for x in xs:
        if x < 0:
            cont += 1
    return cont

print(countNeg([1,2,3,4,5]) == 0) #True
print(countNeg([1,-1,2,-3,4]) == 2) #True
print(countNeg([2,-2]) == 1) #True
print(countNeg([1,-1]) == 1) #True
print(countNeg([1,-3,-4,3,4,-5]) == 3) #True

File: topic_3/Haskell/1-countNeg.hs
-- Conte os números negativos de uma lista

-- do jeito que eu pensei sem filter
listaNegativa xs = [x | x<-xs, x < 0]
countNeg xs = length (listaNegativa xs)

-- com filter
-- countNeg xs = length (filter(\x -> x < 0) xs)

main = do
    a <- readLn :: IO [Int]
    print $ countNeg a